events/worker.go

```diff
@@ -10,7 +10,12 @@ type EventLocker func(event *Event) locks.Locker
 
 func nopLocker(_ *Event) locks.Locker { return locks.NopLocker() }
 
-func resourceIDLocker(event *Event) locks.Locker { return locks.KeyLocker(event.ResourceID) }
+func resourceIDLocker(event *Event) locks.Locker {
+	if event.ResourceID == "" {
+		return locks.NopLocker()
+	}
+	return locks.KeyLocker(event.ResourceID)
```

Comment:
but that would be the same string if inspect requests come. What i see in python-agent is that it doesn't use a lock in inspect event. And in go-agent it locks the event but in py-agent it locks the instance.